new.go

```diff
@@ -0,0 +1,89 @@
+package logging
+
+import (
+	"context"
+)
+
+type Attr [2]interface{}
+
+type AttrMap map[string]interface{}
+
+type LoggerV2 interface {
+	Info(ctx context.Context, msg string, attrs ...Attr)
+}
+
+func NewLoggerV2(baseLogger BaseLogger) LoggerV2 {
+	return loggerV2{
+		baseLogger: baseLogger,
+	}
+}
+
+type loggerV2 struct {
+	baseLogger BaseLogger
+
+	attrs []Attr
+}
+
+func (l loggerV2) log(ctx context.Context, level Level, msg string, attrs ...Attr) {
+	args := attrsToArgs(l.attrs)
+	args = append(args, attrsToArgs(attrs)...)
+	args = append(args, "msg", msg)
+	l.baseLogger.Log(ctx, level, 1, args...)
+}
+
+func (l loggerV2) Info(ctx context.Context, msg string, attrs ...Attr) {
+	l.log(ctx, InfoLevel, msg, attrs...)
+}
+
+type loggerKeyType struct{}
+
+var loggerKey loggerKeyType = struct{}{}
+
+func WithLogger(ctx context.Context, attrs ...Attr) context.Context {
+	if parentLogger, ok := GetLoggerV2(ctx).(loggerV2); ok {
+		attrs = append(attrs, parentLogger.attrs...)
+	}
+
+	logger := loggerV2{
+		baseLogger: globalLogger,
+		attrs:      attrs,
+	}
+
+	ctx = context.WithValue(ctx, loggerKey, logger)
+
+	return ctx
+}
+
+func GetLoggerV2(ctx context.Context) LoggerV2 {
+	logger, ok := ctx.Value(loggerKey).(LoggerV2)
+	if !ok {
+		return NewLoggerV2(globalLogger)
+	}
+	return logger
+}
+
+func InfoV2(ctx context.Context, msg string, attrs ...Attr) {
```

Comment:
Using verbose function name `InfoV2` just to avoid conflict with existing `Info()` in old interface. 

We will have to deal with function conflict if we are adding new interfaces. But I am open for other options like 
a) completely remove old interface from this library (as this library is not yet published, backward compatibility is not the mandatory consideration IMO)
b) create `v2` version of this library, and fully switching to new interface
c) adding new interface in a sub-package (but i have no idea a good name atm)